File: backend/src/main/java/org/springframework/boot/backend/service/BillMethodService.java
package org.springframework.boot.backend.service;

import org.springframework.boot.backend.entity.BillMethod;

import java.util.List;
import java.util.Optional;

public interface BillMethodService {
    List<BillMethod> getAllBillMethods();
    Optional<BillMethod> getBillMethodById(Long id);
}

File: frontend/src/app/services/attributes/bill-method.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { BillMethod } from '../../interfaces/attributes/bill-method';

@Injectable({
  providedIn: 'root'
})
export class BillMethodService {
  private apiUrl = 'http://localhost:8080/billMethod'; // Update with your backend API URL

  constructor(private http: HttpClient) {}

  getAllBillMethods(): Observable<BillMethod[]> {
    return this.http.get<BillMethod[]>(this.apiUrl);
  }

  getBillMethodById(id: number): Observable<BillMethod> {
    return this.http.get<BillMethod>(`${this.apiUrl}/${id}`);
  }
}
